packages/hyperflux/rollup.config.js

```diff
@@ -0,0 +1,80 @@
+/*
+CPAL-1.0 License
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/ir-engine/ir-engine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+The Original Code is Infinite Reality.
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Infinite Reality team.
+All portions of the code written by the Infinite Reality team are Copyright © 2021-2023 
+Infinite Reality. All Rights Reserved.
+*/
+
+
+import dts from 'rollup-plugin-dts'
+import esbuild from 'rollup-plugin-esbuild'
+import typescript from '@rollup/plugin-typescript'
+// import fs from 'fs'
+// import {resolve} from 'path'
+
+// const __dirname = import.meta.dirname;
+// fs.readdir(resolve(__dirname, 'src'), (err, files) => {
+//   files && files.forEach(file => {
+//     console.log(file);
+//   });
+// });
+
+const productionConfig = {
+  minifyInternalExports: true,
+}
+function customBuild() {
+  return esbuild({minify: true})
+}
+
+const externals = ["react", "react-reconciler", "react-reconciler/constants","ts-matches", "ts-toolbelt", "uuid", "@hookstate/core", "@hookstate/identifiable"]
```

Comment:
wonder if we can import this list from the package.json dependencies object keys

Reply:
This is just to suppress a warning. We don't really need it but I liked having the list.

Reply:
I am working on the vite config right now and for some reason it doesn't auto detect externals so I guess this is needed. Very weird